### DISCUSSION TITLE
Accessing near real time consumption data from the Octopus Mini (can it be done)

### DISCUSSION THREAD

**Discussion by Purclewan** [BottlecapDave/HomeAssistant-OctopusEnergy#1096] [Q&A]
Sorry if I'm asking in the wrong place but all my research keeps bringing me back to BottlecapDave's HA intigration.  

In short, my Octopus Smart Meter decided at the end of August to stop transmitting data back to the DCC/Octopus.  I have been trying to get Octopus to sort it but just keep getting fobbed off.  

My Octopus Home Mini is working fine as are my HA integrations.  From my phone app I can see live use down to the last 5 min and daily use in 30 min slots so obviously the data is there in the mini.   My question is, how do I get at it?  I can't see anything in the Octopus Integration that gives me access to my usage at that level of detail.  

My reason for asking is that I have not had an invoice since August and Octopus are telling me it is very likely going to be some time (months) before they will get it sorted.  In the meantime, I would like some way of extracting my usage in real-time to feed a spreadsheet which in turn takes account of my Off Peak and Peak periods. 

At present, I am typing the data from the Home Mini into my Excel spreadsheet which is a bit tedious.  

I am on IOG, have an ASHP, an EV that I charge daily and Solar and Batteries so keeping tabs on my useage is important.   Octopus tell me this can't be done but If the data can be seen on the app it must surely be available.

**Comment by HLFCode** [comment]
Several questions here.
Probably not the right forum - not sure which is.

> to stop transmitting data back to the DCC/Octopus

Are you by any chance in the [northern region](https://www.smartme.co.uk/#qa:~:text=Communications%20Company%20(DCC).-,North%20South%20Divide,-%2D%20As%20the%20characteristics) served by Arqiva rather than Telefonica?
If so radio comms can be patchy.  I've solved two site's meter comms issue with a passive repeater antenna (let me know if you need more info).

>  I would like some way of extracting my usage

If you get the meter working you can download hh data from their app.
I'm sure someone who has a mini will come along and solve the 5 min data download however you could look at a Home Assistant integration for your PV/Battery system and it will automatically log the import/export power and energy. All data from HA is downloadable.

**Reply by Purclewan** [reply]
Thanks for the reply, I'm in Scotland so don't use the cellular network for my smart meter as far as I understand.  I have been told by Octopus that I may need a wireless booster so an LF passive repeater antenna might be an answer.  Would be interested to know more.

I have my PV/Battery already integrated with HA but to be honest, I am a real newby with HA.  I use node-red for all my automation and data extraction so face a steep learning curve learning how to extract data from HA.

**Reply by craigarthur** [reply]
> Are you by any chance in the [northern region](https://www.smartme.co.uk/#qa:~:text=Communications%20Company%20(DCC).-,North%20South%20Divide,-%2D%20As%20the%20characteristics) served by Arqiva rather than Telefonica? If so radio comms can be patchy. I've solved two site's meter comms issue with a passive repeater antenna (let me know if you need more info).

Do you have any details of what you've used that works, I've tried a couple of solutions that have been patchy at best.

**Comment by BottlecapDave** [comment]
Hello. The integration provides access to your home mini data. You need to enable it on your account when you setup the integration - https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/setup/account/#home-mini

Once enabled, this will provide a range of new sensors - https://bottlecapdave.github.io/HomeAssistant-OctopusEnergy/entities/electricity/#home-minipro-entities - which update as often as once a minute, depending on what you've configured in the integration. Have a look at the entities (and their attributes) as all the data you need should be there, including peak/off peak costs (in the form of additional sensors)

**Reply by Purclewan** [reply]
Many thanks, really appreciate you taking the time to answer.  I have set up the integration and can see ( I believe) all the entities and can see what you mean about the data.  I'm relatively new to HA and a bit of a newbie so need to figure out how to extract the data from the entities in a form that I can feed to Node-Red and Excel.  I'm guessing I need to create some form of HA Automation or Script.

From what I can see the data I need is embeded in the "Current Accumulative Cost Electricity" entity.  Can you confirm that this takes into account my OIG tariff.  

Again many thanks for the answer and your hard work with the integration.  Awsome work.

**Reply by BottlecapDave** [reply]
The cost is based on your current tariff with pending/completed dispatches altering the costs to the off peak rate. It also includes the standing charge in the total, but there's an attribute without that included. 

In terms of how to get it into node red, you'd be best to ask on the community forums as I'm not too familiar with node red. It will involve an automation of some kind though.

**Reply by Purclewan** [reply]
Many thanks.

Purclewan.
